File: 프로그래머스/lv1/131705. 삼총사/삼총사.py
from itertools import combinations

def solution(number):
    
    num = list(combinations(number, 3))
    cnt = 0
    for i in num:
        if sum(i) == 0:
            cnt += 1
    
    # return cnt
    return len([i for i in list(combinations(number, 3)) if sum(i) == 0])

File: 프로그래머스/1/131705. 삼총사/삼총사.java
class Solution {
    
    static int ans;
    public int solution(int[] number) {
        ans = 0;        
        comb(number, 0, 0, 0);
        
        return ans;
    }
    
    static void comb(int[] number, int start, int cnt, int sum) {
        
        if (cnt == 3) {
            
            if (sum == 0) {
                ans++;
            }
            return;
        }
        
        for(int i = start; i < number.length; i++) {
            comb(number, i + 1, cnt + 1, sum + number[i]);
        }
    }
}